src/test/java/SearchServletTest.java

```diff
@@ -288,4 +305,73 @@ public void checkAuthenticationErrorIsReturned() throws IOException {
     Assert.assertTrue(stringWriter.toString().contains(AUTHENTICATION_ERROR_MESSAGE));
     verify(response).setStatus(HttpServletResponse.SC_FORBIDDEN);
   }
+
+  @Test
+  public void paginationNextPage() throws IOException {
+    // This test simulates starting on page 1, then moving to page 2.
+
+    // Add 12 mock receipts to datastore.
+    ImmutableSet<Entity> expectedReceipts = TestUtils.addManyTestReceipts(datastore);
+
+    // Perform doGet - this should retrieve max receipts for page, which is 10.
+    when(request.getParameter("isPageLoad")).thenReturn("true");
+    servlet.doGet(request, response);
```

Comment:
There should be vertical space between L317 and L318 to separate the "arrange" statements from the "act" statements.